server/services/aiOrder.js

```diff
@@ -0,0 +1,131 @@
+module.exports = class AIOrderService {
+    static CREATE_CARRIER_LOOP = "CREATE_CARRIER_LOOP";
+
+    constructor(carrierService, waypointService, starService, starUpgradeService) {
+        this.carrierService = carrierService;
+        this.waypointService = waypointService;
+        this.starService = starService;
+        this.starUpgradeService = starUpgradeService;
+        this.orderHandlers = {
+            CREATE_CARRIER_LOOP: this._handleCreateCarrierLoop.bind(this)
```

Comment:
Does the AI actually need to create a looped carrier?

1. Check if there is a carrier in transit between the two stars, if so do nothing.
2. Check if the source star has a carrier, if so send it to the destination with a drop all order.
3. Check if the destination star has a carrier, if so send it to the source star with a collect all order.
4. If no carriers are on the source, destination or in transit then create a carrier and do point 3.

Do we actually need to go through the hassle of setting up loops? Even with a loop we still have to perform the above checks anyway.

Also, if we don't loop carriers then the AI has the flexibility to reuse carriers for transits between multiple stars.